src/pages/settings/Preferences/PriorityModePage.tsx

```diff
@@ -1,49 +1,46 @@
-import PropTypes from 'prop-types';
 import React, {useCallback} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
 import {withOnyx} from 'react-native-onyx';
-import _, {compose} from 'underscore';
+import type {ValueOf} from 'type-fest';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import ScreenWrapper from '@components/ScreenWrapper';
 import SelectionList from '@components/SelectionList';
 import RadioListItem from '@components/SelectionList/RadioListItem';
 import Text from '@components/Text';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
+import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import Navigation from '@libs/Navigation/Navigation';
 import * as User from '@userActions/User';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 
-const propTypes = {
+type PriorityModePageOnyxProps = {
     /** The chat priority mode */
-    priorityMode: PropTypes.string,
-
-    ...withLocalizePropTypes,
+    priorityMode: OnyxEntry<ValueOf<typeof CONST.PRIORITY_MODE>>;
 };
 
-const defaultProps = {
-    priorityMode: CONST.PRIORITY_MODE.DEFAULT,
-};
+type PriorityModePageProps = PriorityModePageOnyxProps;
 
-function PriorityModePage(props) {
+function PriorityModePage({priorityMode}: PriorityModePageProps) {
+    const {translate} = useLocalize();
     const styles = useThemeStyles();
-    const priorityModes = _.map(_.values(CONST.PRIORITY_MODE), (mode) => ({
+    const priorityModes = Object.values(CONST.PRIORITY_MODE).map((mode) => ({
         value: mode,
-        text: props.translate(`priorityModePage.priorityModes.${mode}.label`),
-        alternateText: props.translate(`priorityModePage.priorityModes.${mode}.description`),
+        text: translate(`priorityModePage.priorityModes.${mode}.label`),
+        alternateText: translate(`priorityModePage.priorityModes.${mode}.description`),
         keyForList: mode,
-        isSelected: props.priorityMode === mode,
+        isSelected: priorityMode === mode,
     }));
 
     const updateMode = useCallback(
-        (mode) => {
-            if (mode.value === props.priorityMode) {
+        (mode: (typeof priorityModes)[number]) => {
```

Comment:
Maybe you could create a `PriorityMode` type and use it here and with `priorityModes`, wdyt?